gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/AbstractZosmfService.java

```diff
@@ -77,15 +77,31 @@ protected String getZosmfServiceId() {
      */
     protected String getAuthenticationValue(Authentication authentication) {
         final String user = authentication.getPrincipal().toString();
-        final String password;
+        final char[] password;
         if (authentication.getCredentials() instanceof LoginRequest) {
             LoginRequest loginRequest = (LoginRequest) authentication.getCredentials();
             password = loginRequest.getPassword();
         } else {
-            password = (String) authentication.getCredentials();
+            password = readPassword(authentication.getCredentials());
+        }
+
+        final byte[] userByteArray = user.getBytes(StandardCharsets.UTF_8);
+        final byte[] credentials = new byte[userByteArray.length + 1 + password.length];
+
+        int j = 0;
+        for (byte b : userByteArray) {
+            credentials[j++] = b;
+        }
+        credentials[j++] = (byte) ':';
+        try {
+            for (char c : password) {
+                credentials[j++] = (byte) c;
+            }
+
+            return "Basic " + Base64.getEncoder().encodeToString(credentials);
+        } finally {
+            Arrays.fill(credentials, (byte) 0);
```

Comment:
what about password array, it doesn't need to be zeroed?

Reply:
Just to leave the message here, we discussed it in person and the password data is being filled-in in the Filter